doc/modules/cross_validation.rst

```diff
@@ -373,63 +373,63 @@ Example of Leave-2-Out on a dataset with 4 samples::
   [0 1] [2 3]
 
 
-Leave-One-Label-Out - LOLO
---------------------------
+Leave-One-Group-Out
+-------------------
 
-:class:`LeaveOneLabelOut` (LOLO) is a cross-validation scheme which holds out
-the samples according to a third-party provided array of integer labels. This
-label information can be used to encode arbitrary domain specific pre-defined
+:class:`LeaveOneGroupOut` is a cross-validation scheme which holds out
+the samples according to a third-party provided array of integer groups. This
+group information can be used to encode arbitrary domain specific pre-defined
 cross-validation folds.
 
 Each training set is thus constituted by all the samples except the ones
-related to a specific label.
+related to a specific group.
 
-For example, in the cases of multiple experiments, LOLO can be used to
-create a cross-validation based on the different experiments: we create
-a training set using the samples of all the experiments except one::
+For example, in the cases of multiple experiments, :class:`LeaveOneGroupOut`
+can be used to create a cross-validation based on the different experiments:
+we create a training set using the samples of all the experiments except one::
 
-  >>> from sklearn.model_selection import LeaveOneLabelOut
+  >>> from sklearn.model_selection import LeaveOneGroupOut
 
   >>> X = [1, 5, 10, 50]
   >>> y = [0, 1, 1, 2]
-  >>> labels = [1, 1, 2, 2]
-  >>> lolo = LeaveOneLabelOut()
-  >>> for train, test in lolo.split(X, y, labels):
+  >>> groups = [1, 1, 2, 2]
+  >>> logo = LeaveOneGroupOut()
+  >>> for train, test in logo.split(X, y, groups):
   ...     print("%s %s" % (train, test))
   [2 3] [0 1]
   [0 1] [2 3]
 
 Another common application is to use time information: for instance the
-labels could be the year of collection of the samples and thus allow
+groups could be the year of collection of the samples and thus allow
 for cross-validation against time-based splits.
 
 .. warning::
 
   Contrary to :class:`StratifiedKFold`,
-  the ``labels`` of :class:`LeaveOneLabelOut` should not encode
+  the ``groups`` of :class:`LeaveOneGroupOut` should not encode
   the target class to predict: the goal of :class:`StratifiedKFold`
   is to rebalance dataset classes across
   the train / test split to ensure that the train and test folds have
   approximately the same percentage of samples of each class while
-  :class:`LeaveOneLabelOut` will do the opposite by ensuring that the samples
-  of the train and test fold will not share the same label value.
+  :class:`LeaveOneGroupOut` will do the opposite by ensuring that the samples
+  of the train and test fold will not share the same group value.
```

Comment:
This warning was relevant with the old name, because people would make the confusion. Now, if anything, I think it might be *more* confusing to keep it, WDYT?